wip-0009.md

```diff
@@ -0,0 +1,83 @@
+
+
+## Motivation and Rationale
+
+after more than 3 months of the Witnet mainnet launch, the network has faced episodes of a bigger transaction latency than expected. In periods with a mempool full of transactions, blocks not using the complete space appeared in the Witnet network. This reduces the efficiency of the system, thus increasing indirectly the transaction latency perceived by users. Under a more critical scenario, this problem could pose an availability threat in terms of a [denial-of-service attack][DoS], in which a large number of malicious nodes could be mining empty blocks or, even, endanger the network stability by affecting internal network metrics (e.g. Active Reputation Set).
+
+A thorough investigation concluded that the majority of these blocks (if not all) were produced by non-reputed and non-active nodes (i.e. newcomers to the network). Regardless of the intentionality of this behaviour, these blocks were appearing in a much higher volume than the newcomer mining probability that was designed in the protocol.
+
+The current mining algorithm can be summarized as follows:
+
+1. Miners build a candidate block.
+2. Miners compute their eligibility by using the [VRF]-based [RandPoE] algorithm with by using the ARS census as the difficulty parameter (labelled as N).
+3. If eligible (and no better candidate was received), miners will broadcast their candidate blocks.
+
+Upon reception of multiple candidates, the winner will be selected as follows:
+1. Candidate blocks from reputed nodes are prioritized over those from non-reputed nodes.
+2. If there are still multiple candidates, the one with the lowest VRF is chosen.
+
+The main problem with the aforementioned algorithm is that the eligibility is computed using the ARS as census, but the prioritization is only taking into account the reputed nodes. This led to the increase of the newcomer mining probability. For example, the 18th of January the newcomer mining probability could be computed with the following network conditions:
+
+```
+TRS(Total Reputation Set)  = 2000 nodes
+ARS(Active Reputation Set) = 6000 nodes
+RF(Replication Factor) = 3
+
+P(non-reputed node mines a block) = P(none reputed node mines a block) = P(single reputed does not mine a block)^TRS = (1- P(single reputed mine a block))^TRS = (1 - RF/ARS)^TRS ~= 37%
+```
+
+In the initial mining algorithm design [rand-poe], DoS attacks were carefully taken into account. An upper bound for the probability of an attacker censoring blocks was set to approximately 2%. However, before the mainnet launch some mining parameters were changed to achieve a more fair block reward distribution. It was assummed that the reputation expiration would be slower compared to the activity period (related to the ARS). This assumption did not hold in mainnet as seen in the former example. This mistake led to a significant increase in the newcomer mining probability.
```

Comment:
```suggestion
In the initial mining algorithm design [rand-poe], DoS attacks were carefully taken into account. An upper bound for the probability of an attacker censoring blocks was set to approximately 2%. However, at a certain point before the mainnet launch some mining parameters were changed to achieve a fairer distribution of block rewards. This change assumed that the reputation expiration would be slower compared to the reputation activity period—which has a big influence in considering whether an identity belongs in the ARS. This assumption did not hold in mainnet as seen in the example above. This mistake led to a significant increase in the newcomer mining probability.
```

Reply:
For future reference, let's try to link the PR that made RandPoE deviate from its original design.